docs/DafnyRef/Types.md

```diff
@@ -4000,19 +3996,19 @@ predicate Pos#[_k: nat](s: Stream<int>)
 ```
 
 Some restrictions apply. To guarantee that the greatest fix-point always
-exists, the (implicit functor defining the) co-predicate must be
+exists, the (implicit functor defining the) greatest predicate must be
 monotonic. This is enforced by a syntactic restriction on the form of the
-body of co-predicates: after conversion to negation normal form (i.e.,
+body of greatest predicates: after conversion to negation normal form (i.e.,
 pushing negations down to the atoms), intra-cluster calls of
-co-predicates must appear only in _positive_ positions—that is, they must
+greatest predicates must appear only in _positive_ positions—that is, they must
 appear as atoms and must not be negated. Additionally, to guarantee
 soundness later on, we require that they appear in _co-friendly_
```

Comment:
The issue of "co-friendly" is not the same now as it was when the Leino & Moskal paper was first published. Instead, Dafny allows two kinds of extreme predicates and extreme lemmas, indicated with `[nat]` and `[ORDINAL]` (where `[ORDINAL]` is the default). This indication says which type to use for the `_k` variable. If the type used is `ORDINAL` (which is the default), then the "co-friendly" restriction mentioned above does not apply. (By the way, "continuous" is a better name than "co-friendly".) If the type of `_k` is indicated to be `nat`, then the predicates defined are required to be continuous.